### ISSUE TITLE
Inconvenient suppressing of the needless_continue lint

### ISSUE THREAD

**Issue by stanislav-tkach** [rust-lang/rust-clippy#16256]
### Summary

In order to suppress the `needless_continue` warning the corresponding `allow` attribute must be placed on entire function. I'm not sure if it is easy (or possible) to fix, but ideally the attribute can be placed on the `continue` expression itself. Please see the [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=7a27dcabb149d19cd7707eaed57dc6c0).

### Reproducer

Code:

```rust
#![warn(clippy::pedantic)]

fn some_condition() -> bool { true }
fn another_condition() -> bool { true }

fn main() {
    for _ in 0..5 {
        if some_condition() {
            // ...
            continue;
        }
        
        if another_condition() {
            // ...
            #[allow(clippy::needless_continue)]
            continue;
        }
    }
}
```

Current output:
> warning: this `continue` expression is redundant

Desired output:
No warning as it should be suppressed by the `allow` attribute.

### Version

```text
rustc 1.89.0 (29483883e 2025-08-04)
binary: rustc
commit-hash: 29483883eed69d5fb4db01964cdf2af4d86e9cb2
commit-date: 2025-08-04
host: aarch64-apple-darwin
release: 1.89.0
LLVM version: 20.1.7
```
